acls.go

```diff
@@ -187,6 +198,111 @@ func (h *Headscale) generateACLRules() ([]tailcfg.FilterRule, error) {
 	return rules, nil
 }
 
+func (h *Headscale) generateSSHRules() ([]*tailcfg.SSHRule, error) {
+	rules := []*tailcfg.SSHRule{}
+
+	if h.aclPolicy == nil {
+		return nil, errEmptyPolicy
+	}
+
+	machines, err := h.ListMachines()
+	if err != nil {
+		return nil, err
+	}
+
+	acceptAction := tailcfg.SSHAction{
+		Message:                  "",
+		Reject:                   false,
+		Accept:                   true,
+		SessionDuration:          0,
+		AllowAgentForwarding:     false,
+		HoldAndDelegate:          "",
+		AllowLocalPortForwarding: true,
+	}
+
+	rejectAction := tailcfg.SSHAction{
+		Message:                  "",
+		Reject:                   true,
+		Accept:                   false,
+		SessionDuration:          0,
+		AllowAgentForwarding:     false,
+		HoldAndDelegate:          "",
+		AllowLocalPortForwarding: false,
+	}
+
+	for index, sshACL := range h.aclPolicy.SSHs {
+		action := rejectAction
+		switch sshACL.Action {
+		case "accept":
+			action = acceptAction
+		case "check":
+			checkAction, err := sshCheckAction(sshACL.CheckPeriod)
+			if err != nil {
```

Comment:
In the error case we leave action in the reject state and continue processing the rule. Perhaps it would be better to continue to the next rule instead?

Reply:
Something like this?
```go
var action tailcfg.SSHAction

switch sshACL.Action {
case "accept":
	action = sshAcceptAction()
case "check":
	checkAction, err := sshCheckAction(sshACL.CheckPeriod)
	if err != nil {
		log.Error().
			Msgf("Error parsing SSH %d, check action with unparsable duration '%s'", index, sshACL.CheckPeriod)
	} else {
		action = *checkAction
	}
	fallthrough
default:
	action = sshRejectAction()

	log.Error().
		Msgf("Error parsing SSH %d, unknown action '%s'", index, sshACL.Action)

	return nil, err
}
```

Reply:
No, that fallthrough is wrong. There needs to be a case for `"reject"` as well.

Reply:
From my understanding of [the docs](https://tailscale.com/kb/1193/tailscale-ssh/#ensure-tailscale-ssh-is-permitted-in-acls), only `accept` or `check` is valid. Happy to be proven wrong as I haven't read all the available documentation :-)

Reply:
Oh, that’s right. I had forgotten.